functions/snippets/src/test/java/SnippetsTests.java

```diff
@@ -159,4 +159,10 @@ public void helloBackgroundTest() throws IOException {
     new HelloBackground().helloBackground(request, response);
     assertThat(responseOut.toString(), containsString("Hello John!"));    
   }
+
+  @Test
+  public void envTest() throws IOException {
+    new Env().env(request, response);
+    assertThat(responseOut.toString(), containsString(""));
```

Comment:
Checking for any empty string doesn't actually verify that this function works - any non-null string will contain and empty string.

Reply:
I'm not sure how to set an env var in a test and tried doing so.
Do you know how to set an env var?

Reply:
Can you stub it with Mockito? 

```java
when(System.getenv("FOO")).thenReturn("someTestValue");
```

Reply:
OK. Thanks.

Reply:
Ok, it looks like we'll have to add an additional dependency to do this: 
```xml
<dependency>
  <groupId>com.github.stefanbirkner</groupId>
  <artifactId>system-rules</artifactId>
  <version>1.19.0</version>
  <scope>test</scope>
</dependency>
```

```java
import org.junit.contrib.java.lang.system.EnvironmentVariables;

public void EnvironmentVariablesTest {
  @Rule
  public final EnvironmentVariables environmentVariables
    = new EnvironmentVariables();

  @Test
  public void setEnvironmentVariable() {
    environmentVariables.set("name", "value");
    assertEquals("value", System.getenv("name"));
  }
}
``` 

Source: https://stackoverflow.com/a/35393623/9414803